#!/bin/sh -e
## 140_gold_pr10979_1.dpatch
##
## DP: Description: PR gold/10979: Fix internal errors

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

2009-12-30  Ian Lance Taylor  <iant@google.com>

	PR 10979
	* layout.cc (Layout::relaxation_loop_body): If -Ttext was used,
	don't put the file header and segment headers in the text
	segment.

@DPATCH@
===================================================================
RCS file: /cvs/src/src/gold/layout.cc,v
retrieving revision 1.154
retrieving revision 1.155
diff -u -r1.154 -r1.155
--- src/gold/layout.cc	2009/12/31 03:48:46	1.154
+++ src/gold/layout.cc	2009/12/31 05:43:29	1.155
@@ -1482,6 +1482,12 @@
       != General_options::OBJECT_FORMAT_ELF)
     load_seg = NULL;
 
+  // If the user set the address of the text segment, that may not be
+  // compatible with putting the segment headers and file headers into
+  // that segment.
+  if (parameters->options().user_set_Ttext())
+    load_seg = NULL;
+
   gold_assert(phdr_seg == NULL
 	      || load_seg != NULL
 	      || this->script_options_->saw_sections_clause());
